### PR TITLE
Enable VFS testing in more tests

### PR DESCRIPTION
**Pull request by erikjv**

### CODE REVIEW

**Comment by erikjv** [comment]
Please check what should happen in `moveFileToDifferentFolderOnBothSides()`, because the test passes, but I have no idea what the 2 `PUT` requests contain...

**Comment by TheOneRing** [comment]
Maybe metadata changes

**Comment by TheOneRing** [comment]
> Please check what should happen in `moveFileToDifferentFolderOnBothSides()`, because the test passes, but I have no idea what the 2 `PUT` requests contain...

If  you understand the difference please add a short comment to explain it.

File: test/testutils/syncenginetestutils.cpp [hunk 1]
```diff
@@ -920,16 +944,15 @@ QNetworkReply *FakeQNAM::createRequest(QNetworkAccessManager::Operation op, cons
     return reply;
 }
 
-FakeFolder::FakeFolder(const FileInfo &fileTemplate, OCC::Vfs::Mode vfsMode)
```
Comment (erikjv) [inline test/testutils/syncenginetestutils.cpp:923]: @TheOneRing can you verify that this constructor is doing the correct thing, including wrt. the call to `switchVfs` and the surrounding code?

File: test/testsyncmove.cpp [hunk 2]
```diff
@@ -188,25 +210,35 @@ private slots:
...
+        counter.reset();
 
-        // Move-and-change, causing a upload and delete
+        // Move-and-change, mtime+size, causing a upload and delete
+        QVERIFY(fakeFolder.currentLocalState().find("A/a2")->isDehydratedPlaceholder == filesAreDehydrated); // no-one touched it, so the hydration state should be the same as the initial state
         fakeFolder.localModifier().rename("A/a2", "A/a2m");
-        fakeFolder.localModifier().appendByte("A/a2m");
+        fakeFolder.localModifier().setContents("A/a2m", 'x', fakeFolder.remoteModifier().contentSize + 1);
         QVERIFY(fakeFolder.syncOnce());
+        QVERIFY(!fakeFolder.currentLocalState().find("A/a2m")->isDehydratedPlaceholder); // We overwrote all data in the file, so whatever the state was before, it is no longer dehydrated
```
Comment (erikjv) [inline test/testsyncmove.cpp:224]: This fails for winvfs

**Comment by TheOneRing** [comment]
I just changed the target to master as 2.10 is more or less closed by now.
Can you have a look at conflicts?

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/11955/6/1

<details>


</details>

File: test/testchunkingng.cpp [hunk 3]
```diff
@@ -62,25 +65,27 @@ private slots:
 
     void testFileUpload() {
         FakeFolder fakeFolder{FileInfo::A12_B12_C12_S12()};
+        fakeFolder.syncEngine().account()->setCapabilities({ { "dav", QVariantMap{ {"chunking", "1.0"} } } });
```
Comment (erikjv) [inline test/testchunkingng.cpp:68]: @TheOneRing is this not needed anymore?

Comment (TheOneRing) [reply test/testchunkingng.cpp:68]: yes it was removed see https://github.com/owncloud/client/blob/master/test/testutils/testutils.cpp#L35 and https://github.com/owncloud/client/blob/master/test/testutils/testutils.cpp#L35

File: test/testutils/syncenginetestutils.cpp [hunk 4]
```diff
@@ -41,65 +41,123 @@ PathComponents PathComponents::subComponents() const &
...
+{
+    if (_processArguments.isEmpty()) {
+        return true;
+    }
+
+    HelperProcess helper(_rootDir, _processArguments);
+    helper.waitForStarted();
+    _processArguments.clear(); // the helper has a copy, we clean it now for the next run
+    do {
+        QThread::msleep(50);
```
Comment (TheOneRing) [inline test/testutils/syncenginetestutils.cpp:126]: please use 50ms, as QThread has no overload for that  std::this_thread::sleep_for(50ms)
```suggestion
        std::this_thread::sleep_for(50ms);
```

File: test/testutils/syncenginetestutils.cpp [hunk 5]
```diff
@@ -41,65 +41,123 @@ PathComponents PathComponents::subComponents() const &
...
+    // Non-classic mode:
+
+    HelperProcess p(_rootDir, _processArguments);
+    _processArguments.clear();
+
+    do {
+        // process any pending events
+        QThread::currentThread()->eventDispatcher()->processEvents(QEventLoop::AllEvents);
+        // wait a bit to get the helper started with it's work
+        QThread::msleep(50);
```
Comment (TheOneRing) [inline test/testutils/syncenginetestutils.cpp:150]:
```suggestion
        std::this_thread::sleep_for(50ms);
```

File: test/testutils/test_helper.cpp [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+
+    const QString _fileName;
+};
+
+Command::~Command() { }
+
+class SetMtimeCommand : public Command
+{
+public:
+    static const QString name;
```
Comment (TheOneRing) [inline test/testutils/test_helper.cpp:54]: Hmm I guess we could use `constexpr std::string_view name` then the name could be inline?
(I don't like the name to be defined outside of the class).

File: test/testutils/test_helper.cpp [hunk 7]
```diff
@@ -0,0 +1,475 @@
...
+        cerr << qPrintable(name) << endl;
+        QFile f(rootDir.filePath(_fileName));
+        if (!f.open(QIODevice::WriteOnly | QIODevice::Append)) {
+            cerr << "Error: cannot open file '" << qPrintable(_fileName) << "' for " << qPrintable(name) << " command: "
+                 << qPrintable(f.errorString()) << endl;
+            return false;
+        }
+        char ch = _ch;
+        if (ch == '\0') {
+            ch = f.readAll().at(0);
```
Comment (TheOneRing) [inline test/testutils/test_helper.cpp:231]: f.read(1)?

**Review by TheOneRing — COMMENTED** [review]
Could it make sense to switch from positional parsing to json in a follow up?
In my opinion the mental load for parsing the parsers might be less with json :D

File: test/testallfilesdeleted.cpp [hunk 8]
```diff
@@ -71,6 +71,7 @@ private slots:
         for (auto it = children.cbegin(); it != children.cend(); ++it) {
             modifier.remove(it.key());
         }
+        QVERIFY(fakeFolder.applyLocalModificationsWithoutSync());
```
Comment (TheOneRing) [inline test/testallfilesdeleted.cpp:74]: That was previously done implicit? Or why is it needed now.

File: test/testutils/syncenginetestutils.cpp (hunk 5 above)
Comment (erikjv) [reply test/testutils/syncenginetestutils.cpp:150]: done

File: test/testutils/test_helper.cpp (hunk 6 above)
Comment (erikjv) [reply test/testutils/test_helper.cpp:54]: done

File: test/testutils/test_helper.cpp (hunk 7 above)
Comment (erikjv) [reply test/testutils/test_helper.cpp:231]: removed the read completely

File: test/testallfilesdeleted.cpp (hunk 8 above)
Comment (erikjv) [reply test/testallfilesdeleted.cpp:74]: added a comment

**Pull request merged**